pkg/operator/apis/monitoring/v1/http_types.go

```diff
@@ -20,40 +20,154 @@ import (
 	"net/url"
 
 	"github.com/prometheus/common/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/GoogleCloudPlatform/prometheus-engine/pkg/secrets"
 )
 
-// Auth sets the `Authorization` header on every scrape request.
-//
-// Currently the credentials are not configurable and always empty.
+// SecretSelector represents a reference to a secret from the given provider
+// e.g. Kubernetes Secret. Only one provider can be used at the same time.
+type SecretSelector struct {
+	// KubernetesSecret represents reference to a given key from certain Kubernetes Secret
+	// in a given namespace.
+	// +optional
+	KubernetesSecret *KubernetesSecretKeySelector `json:"secret,omitempty"`
```

Comment:
```suggestion
	Secret *KubernetesSecretKeySelector `json:"secret,omitempty"`
```

Let's keep the type explicit, but here we could keep it same as field name to avoid confusion. We are configuring Kubernetes resource so "secret" meaning Kubernetes secret makes sense.

Notably, the API description is misleading (secret vs KubernetesSecret) 

<img width="567" alt="image" src="https://github.com/GoogleCloudPlatform/prometheus-engine/assets/6950331/4b8645cc-d751-4516-b45f-e010d6089599">